diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -471,89 +519,80 @@ class ASTNodeUtilsTest {
                 * @param a - dummy int
                 */
-                fun foo(a: Int): Int = 2 * a
+                public fun foo(a: Int): Int = 2 * a
             }
         """.trimIndent()
+        var secondCounter = 0
         applyToCode(code){node ->
-            if (node.elementType == MODIFIER_LIST)
+            if (node.elementType == MODIFIER_LIST) {
                 Assert.assertTrue(node.isAccessibleOutside())
+                secondCounter++
+            }
         }
+        Assert.assertEquals(1, firstCounter)
+        Assert.assertEquals(1, secondCounter)
     }
     @Test
     fun `test leaveOnlyOneNewLine`() {
         val code = """
             var x = 2
-            
-            
+
+
         """.trimIndent()
+        var counter = 0
         applyToCode(code) { node ->
             if (node.elementType == WHITE_SPACE && node.text.contains("\n\n")) {
                 val parent = node.treeParent
                 val firstText = node.text
                 node.leaveOnlyOneNewLine()
                 var secondText: String? = null
                 parent.getChildren(null).forEach { newNode -> secondText = newNode.text }
-                Assert.assertNotEquals(firstText, secondText)
+                Assert.assertEquals("\n", secondText)
+                Assert.assertEquals("\n\n", firstText)
+                counter++
             }
         }
+        Assert.assertEquals(1, counter)
     }
 
     @Test
-    fun `moveChildBefore 1 - Should correctly move node child before another`() {
+    fun `moveChildBefore 1 - reverse`() {
+        var counter = 0
         applyToCode("""
                 |val a = 0
                 |val b = 1
             """.trimMargin()) { node ->
-            if (node.elementType == CLASS_BODY) {
-                val val1 = node.getFirstChildWithType(PROPERTY)
-                val val2 = val1!!.nextSibling { it.elementType == PROPERTY }!!
-                val whiteSpace = val1.treeNext.clone() as ASTNode
-                node.moveChildBefore(val2, val1, false)
-                node.addChild(whiteSpace, val1)
-                node.removeChild(val1.treeNext)
-                Assert.assertTrue(node.text == """
-                    |val b = 1
-                    |val a = 0
-                    |
-                    """.trimMargin()
-                )
+            val listBeforeMove = mutableListOf<IElementType>()
+            val listAfterMove = mutableListOf<IElementType>()
+            if(node.getChildren(null).isNotEmpty()){
+                node.getChildren(null).forEach { listBeforeMove.add(it.elementType) }
+                node.getChildren(null).forEachIndexed { index, astNode ->
+                    node.moveChildBefore(astNode, node.getChildren(null)[node.getChildren(null).size - index -1])
+                }
+                node.getChildren(null).forEach { listAfterMove.add(it.elementType) }
+                Assert.assertEquals(listBeforeMove, listAfterMove.reversed())
```

Comment:
Also, nodes are removed from AST, but objects are still present in heap. You could compare collections of nodes directly, without leaving just elementType.

Reply:
Yes, nodes are replaced with new ones, so nodes cannot be compared here. Leave as is.